scripts/with-go-mod.sh

```diff
@@ -6,28 +6,40 @@
 # when the command is finished. This script should be dropped when this
 # repository is a proper Go module with a permanent go.mod.
 
-set -e
+set -euo pipefail
 
 SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 ROOTDIR="$(cd "${SCRIPTDIR}/.." && pwd)"
 
-if test -e "${ROOTDIR}/go.mod"; then
-	{
-		scriptname=$(basename "$0")
-		cat >&2 <<- EOF
-			$scriptname: WARN: go.mod exists in the repository root!
-			$scriptname: WARN: Using your go.mod instead of our generated version -- this may misbehave!
-		EOF
-	} >&2
-else
+cleanup_paths=""
+create_symlink() {
+	local target="$1"
+	local link="$2"
+
+	pushd "${ROOTDIR}" > /dev/null
+	trap 'popd > /dev/null' RETURN
```

Comment:
Whoa whoa whoa, what?  This is wild, and unnecessary -- there are either flags on `ln` that can DTRT or `cp --symbolic-link`.  If you can't get the arguments finagled correctly, let me know and I'll happily help sort it out further.

Reply:
❤️ I ran into the script not detecting that my own (relative) symlink was same as the symlink it wanted to create, so thought that changing to the directory would be the best way to "create a relative symlink here' and make it detect that my own symlink was "same" (so no warning printed) 🙈

Reply:
So maybe something like https://superuser.com/a/196655/101945? :eyes:

Reply:
OK; I think I found why it originally didn't work; I used an absolute path for both sides; now using absolute path for only the "source" side, and that looks to work; creates relative symlink, and detects my own symlink for being "ok"

Reply:
> https://superuser.com/a/196655/101945

Oh! GitHub didn't show your comment; that looks interesting as well yes; let me give that a try as well!

Reply:
If you use absolute path on both sides, you can use `ln --relative` to have `ln` automatically "fix" it for you.

Reply:
Yeah; so ... macOS and BSD 🙈 - don't think they have that option 🫠

Reply:
!["I'm deleting them from my life." webcomic cover](https://github.com/user-attachments/assets/ebe10504-186c-4b2c-bd80-5d60455f9df5)